code/core/system/proxy-velodyne16/src/velodyne16Decoder.cpp

```diff
@@ -155,49 +173,62 @@ Velodyne16Decoder::Velodyne16Decoder(odcore::io::conference::ContainerConference
     , m_segment(NULL)
     , m_velodyneContainer(c)
     , m_spc()
-    , m_calibration("")
+    , m_calibration(s)
     , m_withSPC(false)
     , m_withCPC(true)
     , m_startAzimuth(0.0)
-    , m_distanceStringStream("")
+    , m_distanceStringStreamNoIntensity("")
+    , m_distanceStringStreamWithIntensity("")
     , m_isStartAzimuth(true) {
-    
-    initializeArraysCPC();
+        
+    index16sensorIDs();
 }
 
 Velodyne16Decoder::~Velodyne16Decoder() {
-    if(m_withSPC){
+    if (m_withSPC) {
         free(m_segment);
     }
 }
 
 //Update the shared or compact point cloud when a complete scan is completed.
 void Velodyne16Decoder::sendPointCloud() {  
     //Send shared point cloud
-    if(m_withSPC){
+    if (m_withSPC) {
         if (m_velodyneSharedMemory->isValid()) {
             Lock l(m_velodyneSharedMemory);
             memcpy(m_velodyneSharedMemory->getSharedMemory(), m_segment, m_SIZE);
             //Set the size and width of the shared point cloud of the current frame
             m_spc.setSize(m_SIZE); // Size in raw bytes.
             m_spc.setWidth(m_pointIndexSPC); // Number of points.
             Container c(m_spc);
```

Comment:
You need to set the sample time point for all containers to be sent to the same value; otherwise, the receiver could think that these sensor value are from various time points. I suggest that you store a TimeStamp when you receive the first byte from Velodyne for the current sweep and then, you use this time stamp for the containers by calling setSampleTimeStamp(). Then, the time stamping is identical for all various options per Velodyne sweep.